src/lib/support/tests/TestSpan.cpp

```diff
@@ -29,24 +29,83 @@
 
 using namespace chip;
 
-static void SpanConstructors(nlTestSuite * inSuite, void * inContext)
+static void TestByteSpan(nlTestSuite * inSuite, void * inContext)
 {
     uint8_t arr[] = { 1, 2, 3 };
-    Span<uint8_t> s1(arr, 2);
 
+    ByteSpan s0 = ByteSpan();
+    NL_TEST_ASSERT(inSuite, s0.data() == nullptr);
+    NL_TEST_ASSERT(inSuite, s0.size() == 0);
+    NL_TEST_ASSERT(inSuite, s0.empty());
+    NL_TEST_ASSERT(inSuite, s0.data_equal(s0));
+
+    ByteSpan s1(arr, 2);
+    NL_TEST_ASSERT(inSuite, s1.data() == arr);
+    NL_TEST_ASSERT(inSuite, s1.size() == 2);
+    NL_TEST_ASSERT(inSuite, !s1.empty());
+    NL_TEST_ASSERT(inSuite, s1.data_equal(s1));
+    NL_TEST_ASSERT(inSuite, !s1.data_equal(s0));
+
+    ByteSpan s2(arr);
+    NL_TEST_ASSERT(inSuite, s2.data() == arr);
+    NL_TEST_ASSERT(inSuite, s2.size() == 3);
+    NL_TEST_ASSERT(inSuite, s2.data()[2] == 3);
+    NL_TEST_ASSERT(inSuite, !s2.empty());
+    NL_TEST_ASSERT(inSuite, s2.data_equal(s2));
+    NL_TEST_ASSERT(inSuite, !s2.data_equal(s1));
+
+    ByteSpan s3 = s2;
+    NL_TEST_ASSERT(inSuite, s3.data() == arr);
+    NL_TEST_ASSERT(inSuite, s3.size() == 3);
+    NL_TEST_ASSERT(inSuite, s3.data()[2] == 3);
+    NL_TEST_ASSERT(inSuite, !s3.empty());
+    NL_TEST_ASSERT(inSuite, s3.data_equal(s2));
+
+    uint8_t arr2[] = { 3, 2, 1 };
+    ByteSpan s4(arr2);
+    NL_TEST_ASSERT(inSuite, !s4.data_equal(s2));
+}
```

Comment:
There should be a test here that an empty span with a buffer ptr and one without a buffer ptr are data_equal, since that came up explicitly in the PR review as a problem....

Reply:
added test